homework/src/app.js

```diff
@@ -1,11 +1,157 @@
+
+//Now add another favorite animal to the array but make sure it is placed after 'blowfish' and before 'capricorn'.
+favoriteAnimals.splice(1, 0, 'meerkat');
+console.log(favoriteAnimals);
+
+//Delete 'giraffe' from the array.
+favoriteAnimals.splice(3, 1); 
+console.log(favoriteAnimals);
+
+ /* OPERATORS */
+
+ // 1. Mathematically, if a == b and b == c, then a == c. Can you explain in your words why this is not true for the above snippet?
+
+var a='0';
+var b= 0;
+var c=[];
+
+
+ console.log( a == b ? 'true':'false');
+ console.log( b == c ? 'true':'false');
+ console.log( a == c ? 'true':'false');
+
+ // a is string and c is array and there value are not equal.
+
+// 2.What do you think will happen with x == y, x === y and z === y and z == x? Try to think of answer first before running it on browser. Also, explain in your words why this happens?
+
+//where x and y also hold an address of the array, so we get false because the values and structure are different from the references they were originally created in the same image. arrays are reference rounds and generated once in memory
+
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+console.log( x == y ? 'true':'false');
+ console.log( x === y ? 'true':'false');
+ console.log( z === y ? 'true':'false');
+ console.log( z == x ? 'true':'false');
+
+// 3.
+
+const o1 = { foo: "bar" };
+const o2 = { foo: "bar" };
+const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+o1.property = 'object';
+console.log(o1);
+console.log(o2);
+console.log(o3);
+// comment:  we change o1 then o3 is not changed.
+
+o2.property='this object';
+console.log(o1);
+console.log(o2);
+console.log(o3);
+//comment:we change o2 then o3 is changed
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+// yes. because we cannot create it will say o3 is undefined.
+
+/* QUESTIONS FROM HYF REPOS */
+
+//8.1 First declare at least four variables and assign them different data types.
+var a=4;
+var b="fruit";
+var c=['fish','horse'];
+var d= true;
+
+//8.2 For each variable write a console.log statement that logs the value
+console.log("The value of my variable a is: "+a);
+console.log("The value of my variable a is: "+b);
+console.log("The value of my variable a is: "+c);
+console.log("The value of my variable a is: "+d);
+
+//8.3 Now write a console.log statement wherein you first explain in words what you think the type of your variables is.
+console.log("a variable is number");
+console.log("b variable is string");
+console.log("c variable is object");
+console.log("d variable is boolean");
+
+//8.4 Now use typeof to log the actual type of your variables.
+console.log(typeof a);
+console.log(typeof b);
+console.log(typeof c);
+console.log(typeof d);
+
+//8.5 Now compare the types of your different variables with one another.
+//8.6 Make sure to also show a message when the variables you are comparing are not the same type.
+
+if(a==b){
```

Comment:
Ah. You shouldn't use `==` operator for comparing two values. Always make use of `===` operators. 🙂 

Also, there seems to be some indentation problems with the code you are writing using your editor. Could you please go through this document for JS2 and install all necessary extensions which would help you write code in a nicer way. https://github.com/HackYourFuture/JavaScript2/blob/master/Week1/README.md

Reply:
yes it must be === .  Thank you for review .